yt/frontends/stream/misc.py

```diff
@@ -0,0 +1,35 @@
+from typing import List
+
+import numpy as np
+
+from yt._typing import DomainDimensions
+
+
+def _validate_cell_widths(
+    cell_widths: List[np.ndarray],
+    domain_dimensions: DomainDimensions,
+) -> List[List[np.ndarray],]:
+    # check dimensionality
+    ndims = len(domain_dimensions)
+    nwids = len(cell_widths)
+    if nwids != ndims:
+        raise ValueError(
+            f"The number of elements in cell_widths ({nwids}) "
+            f"must match the number of dimensions ({ndims})."
+        )
+
+    # check the dtypes for each dimension, upcast to float64 if needed
+    cast_dims = []
+    for idim, cell_wid in enumerate(cell_widths):
+        if cell_wid.dtype != np.float64:
+            cast_dims.append(idim)
+
+    for idim in cast_dims:
+        cell_widths[idim] = cell_widths[idim].astype(np.float64)
```

Comment:
Couldn't this be simplified ?
```suggestion
    for idim in len(cell_widths):
        cell_widths[idim] = cell_widths[idim].astype(np.float64, copy=False)
```

Reply:
yes, yes it could. I think I put it in the loop to avoid calling astype if not needed, but with the copy=False that would alleviate any concern I had (which might not have been well founded to begin with).